homeassistant/components/sensor/citybikes.py

```diff
@@ -0,0 +1,186 @@
+CONF_NETWORK = 'network'
+CONF_RADIUS = 'radius'
+CONF_STATIONS_LIST = 'stations'
+ATTR_STATION_ID = 'id'
+ATTR_STATION_UID = 'id'
+ATTR_STATION_NAME = 'name'
+ATTR_EXTRA = 'extra'
+ATTR_EMPTY_SLOTS = 'empty_slots'
+ATTR_FREE_BIKES = 'free_bikes'
+ATTR_TIMESTAMP = 'timestamp'
+CITYBIKES_ATTRIBUTION = "Information provided by the CityBikes Project "\
+                        "(https://citybik.es/#about)"
+
+
+PLATFORM_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CONF_RADIUS, CONF_STATIONS_LIST),
+    PLATFORM_SCHEMA.extend({
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_NETWORK): cv.string,
+        vol.Inclusive(CONF_LATITUDE, 'coordinates'): cv.latitude,
+        vol.Inclusive(CONF_LONGITUDE, 'coordinates'): cv.longitude,
+        vol.Exclusive(CONF_RADIUS, 'station_filter'): cv.positive_int,
+        vol.Exclusive(CONF_STATIONS_LIST, 'station_filter'):
+            vol.All(
+                cv.ensure_list,
+                vol.Length(min=1),
+                [cv.string])
+    }))
+
+
+def _filter_stations(network, radius, latitude, longitude, stations_list):
+    if radius > 0:
+        for station, dist in network.stations.near(latitude, longitude):
+            # 'dist' is in weird units, let's calculate again
+            dist = location.distance(latitude, longitude,
+                                     station[ATTR_LATITUDE],
+                                     station[ATTR_LONGITUDE])
+            if dist > radius:
+                break
+            yield station
+    else:
+        for station in network.stations:
+            if station['id'] in stations_list:
+                yield station
+                continue
+            if 'extra' in station:
+                if 'uid' in station['extra']:
+                    if str(station['extra']['uid']) in stations_list:
+                        yield station
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the CityBikes platform."""
+    from citybikes import Client, Network
+
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+
+    network_uid = config.get(CONF_NETWORK)
+    if not network_uid:
+        network = Client().networks.near(latitude, longitude)[0][0]
+    else:
+        network = Network(Client(), uid=network_uid)
+
+    stations_list = config.get(CONF_STATIONS_LIST, [])
+    radius = config.get(CONF_RADIUS, 0)
+    if not hass.config.units.is_metric:
+        radius = distance.convert(radius, LENGTH_FEET, LENGTH_METERS)
+    poller = CityBikesNetworkPoller(network)
+
+    add_devices(CityBikesStationSensor(poller, station,
+                                       config.get(CONF_NAME, DOMAIN))
+                for station in _filter_stations(network, radius, latitude,
+                                                longitude, stations_list))
+
+
+class CityBikesNetworkPoller(object):
```

Comment:
I'm used to inheriting from `object` :)
I also looked at other components, and saw it being done throughout the code, so thought it's acceptable.